web/js/MapUI/Components/UpdateOpacity/UpdateOpacity.js

```diff
@@ -0,0 +1,109 @@
+import { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {
+  each as lodashEach,
+  find as lodashFind,
+} from 'lodash';
+import { getActiveLayers } from '../../../modules/layers/selectors';
+import * as layerConstants from '../../../modules/layers/constants';
+
+const UpdateOpacity = (props) => {
+  const {
+    action,
+    activeLayers,
+    activeString,
+    compare,
+    findLayer,
+    isCompareActive,
+    ui,
+    updateLayerVisibilities,
+  } = props;
+
+  useEffect(() => {
+    if (action.type === layerConstants.UPDATE_OPACITY) {
+      updateOpacity(action);
+    }
+  }, [action]);
+
+  const updateGranuleLayerOpacity = (def, activeString, opacity, compare) => {
```

Comment:
activeString & compare are already declared (lines 15 & 16). Can we rename these parameters to avoid confusion?

Reply:
This is a good catch but this is another one that deals with granules that I haven't really been able to test fully yet. I am certain that I will have to revisit this function soon enough so I am just going to keep these there for now and I will remove them if necessary when I am able to test.